compaction/time_window_compaction_strategy.cc

```diff
@@ -184,16 +184,23 @@ class classify_by_timestamp {
     };
 };
 
-uint64_t time_window_compaction_strategy::adjust_partition_estimate(const mutation_source_metadata& ms_meta, uint64_t partition_estimate) const {
-    if (!ms_meta.min_timestamp || !ms_meta.max_timestamp) {
-        // Not enough information, we assume the worst
-        return partition_estimate / max_data_segregation_window_count;
+uint64_t time_window_compaction_strategy::adjust_partition_estimate(const mutation_source_metadata& ms_meta, uint64_t partition_estimate, schema_ptr s) const {
+    // If not enough information, we assume the worst
+    auto estimated_window_count = max_data_segregation_window_count;
+    auto default_ttl = std::chrono::duration_cast<std::chrono::seconds>(s->default_time_to_live());
+    bool min_and_max_ts_available = ms_meta.min_timestamp && ms_meta.max_timestamp;
+
+    if (!min_and_max_ts_available && default_ttl.count()) {
+        std::chrono::seconds window_size = _options.get_sstable_window_size();
+
+        estimated_window_count = (default_ttl.count() + window_size.count() - 1) / window_size.count();
```

Comment:
It's safer to divide duration by duration. I see you divide seconds.count by seconds.count, but if you write it as (default_ttl + window_size - 1s) / window_size, the units cancel out by themselves.

Reply:
you can simplify the computation: compute the time range that the data spans (either from min_max_timestamp or from the default ttl) and apply the same calculation to the result.